frontend/controller/serviceworkers/sw-primary.js

```diff
@@ -77,6 +81,13 @@ const setupRootState = () => {
   if (!rootState.notifications) rootState.notifications = Object.create(null)
   if (!rootState.notifications.items) rootState.notifications.items = []
   if (!rootState.notifications.status) rootState.notifications.status = Object.create(null)
+
+  if (!rootState.periodicNotificationAlreadyFiredMap) {
+    rootState.periodicNotificationAlreadyFiredMap = {
+      alreadyFired: Object.create(null), // { notificationKey: boolean },
+      lastRun: Object.create(null) // { notificationKey: number },
+    }
+  }
```

Comment:
Question, below we have:

```js
sbp('okTurtles.events/on', CHELONIA_RESET, setupRootState)
```

But everything inside this function first checks if the thing isn't already defined... so is it really a "reset"..?

Reply:
> Or, do you have a change request that'd simplify things? (When chelonia/reset is called, rootState will be rid of most keys)

Ah, that last parenthetical answers my question, thanks!